File: design-patterns/ts/state/order-status/order-deliver-state.ts
// order-deliver-state.ts

import OrderContext from "./order-context";
import OrderState from "./order-state";

class OrderDeliverState extends OrderState {
    constructor(context: OrderContext) {
        super(context);
    }

    process(): void {
        // Write code to process the order
        console.log("Delivering the order");

        this.context.setState(this.context.getOrderReceiveState());
    }
}

export default OrderDeliverState;

File: design-patterns/go/state/order_state/order_deliver_state.go
// order_deliver_state.go

package main

import "fmt"

type OrderDeliverState struct {
	*OrderState
}

func NewOrderDeliverState(context *OrderContext) (orderDeliverState *OrderDeliverState) {
	orderDeliverState = &OrderDeliverState{}
	orderDeliverState.OrderState = NewOrderState(orderDeliverState, context)
	return
}
func (orderDeliverState *OrderDeliverState) Process() {
	fmt.Println("Delivering the order")
	orderDeliverState.context.SetState(orderDeliverState.context.GetOrderReceiveState())
}
